sickbeard/clients/download_station_client.py

```diff
@@ -31,20 +36,73 @@ def __init__(self, host=None, username=None, password=None):
 
         super(DownloadStationAPI, self).__init__('DownloadStation', host, username, password)
 
-        self.url = self.host + 'webapi/DownloadStation/task.cgi'
+        self.urls = {
+            'login': urljoin(self.host, 'webapi/auth.cgi'),
+            'task': urljoin(self.host, 'webapi/DownloadStation/task.cgi'),
+            'info': urljoin(self.host, '/webapi/DownloadStation/info.cgi'),
+            'dsminfo': urljoin(self.host, '/webapi/entry.cgi')
+        }
 
-    def _get_auth(self):
+        self.url = self.urls['task']
+
+        self.error_map = {
+            100: 'Unknown error',
+            101: 'Invalid parameter',
+            102: 'The requested API does not exist',
+            103: 'The requested method does not exist',
+            104: 'The requested version does not support the functionality',
+            105: 'The logged in session does not have permission',
+            106: 'Session timeout',
+            107: 'Session interrupted by duplicate login'
+        }
+        self.checked_destination = False
+        self.destination = sickbeard.TORRENT_PATH
 
-        auth_url = self.host + 'webapi/auth.cgi?api=SYNO.API.Auth&version=2&method=login&account=' + self.username + '&passwd=' + self.password + '&session=DownloadStation&format=sid'
+    def _check_response(self):
+        """
+        Check if session is valid
+        """
 
         try:
-            self.response = self.session.get(auth_url, verify=False)
-            self.auth = self.response.json()['data']['sid']
-        except Exception:
-            return None
+            jdata = self.response.json()
+        except ValueError:
+            self.session.cookies.clear()
+            self.auth = False
+            return self.auth
+
+        self.auth = jdata.get('success')
+        if not self.auth:
+            error_code = jdata.get('error', {}).get('code')
+            sickbeard.logger.log('{}'.format(self.error_map.get(error_code, jdata)))
+            self.session.cookies.clear()
 
         return self.auth
 
+    def _get_auth(self):
+        if self.session.cookies and self.auth:
+            return self.auth
+
+        params = {
+            'api': 'SYNO.API.Auth',
+            'version': 2,
+            'method': 'login',
+            'account': self.username.encode('utf-8') if isinstance(self.username, unicode) else self.username,
```

Comment:
@miigotu requests already encodes all params to utf-8
https://github.com/kennethreitz/requests/blob/master/requests/models.py#L95